compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -269,30 +299,27 @@ template <typename Config> class MapAllocatorCache {
           OldestTime = Entry.Time;
         Entry = PrevEntry;
       }
-      if (EntriesCount >= MaxCount) {
-        if (IsFullEvents++ == 4U)
-          EmptyCache = true;
-      } else {
-        for (u32 I = 0; I < MaxCount; I++) {
-          if (Entries[I].isValid())
-            continue;
-          if (I != 0)
-            Entries[I] = Entries[0];
-          Entries[0] = Entry;
-          EntriesCount++;
-          if (OldestTime == 0)
-            OldestTime = Entry.Time;
-          EntryCached = true;
-          break;
-        }
+
+      // All excess entries are evicted from the cache
+      while (EntriesCount >= MaxCount) {
+        // Save MemMaps of evicted entries to perform unmap outside of lock
+        EvictionMemMaps.push_back(Entries[LRUTail].MemMap);
+        remove(LRUTail);
```

Comment:
Just to clarify, you want `needToEvict` to decrement the `EntriesCount` variable instead? And if we do that, should something similar be done for `insert`?

Reply:
No, use it as the loop condition.
```
while (needToEvict()) {
  ...
}
```

Reply:
Ok, I wonder if we should pass `MaxCount` to `needToEvict` because we don't want to perform an atomic load of `MaxEntriesCount` for every iteration, right?

Reply:
As you said, it's a rare event. In addition, it's atomic_load_relaxed so it should be a normal load. So we can do the load without performance concern